zdm/cosmology.py

```diff
@@ -274,6 +274,39 @@ def dFnu_to_dEnu(z,alpha=0,bandwidth=1.e9):
 	dE = 4*np.pi*(dl(z))**2/(1.+z)**(2.-alpha)
 	return dE
 
+######### possible source evolution functions go here ##########
+
+def choose_source_evolution_function(which=0):
+	"""
+	Selects which source evolution function to use
+	These are now generalised to take multiple parameters
+	Could implement arbitrary many of these
+	"""
+	if which==0:
+		source_evolution=sfr_evolution
+	elif which==1:
+		source_evolution=opz_evolution
+	else:
+		raise ValueError("Undefined source evolution function ",which," choose 0 or 1")
+	return source_evolution
+
+def sfr_evolution(z,*params):
+	"""
+	Madau & dickenson 2014
+	Takes only a single param
+	"""
+	return (1.0025738*(1+z)**2.7 / (1 + ((1+z)/2.9)**5.6))**params[0]
+	
+
+def opz_evolution(z,*params):
+	"""
+	Same as SFR, but without numerator, i.e. just (1+z)**2.7
```

Comment:
Do you mean without the denominator(instead of the numerator)?

Reply:
Oops, yes, good catch!

Reply:
Fixed...